src/TestSuite/IntegrationTestCase.php

```diff
@@ -434,10 +434,15 @@ protected function _buildRequest($url, $method, $data)
         $session = Session::create($sessionConfig);
         $session->write($this->_session);
         list ($url, $query) = $this->_url($url);
+        $tokenUrl = $url;
+
+        if (!empty($query)) {
```

Comment:
Check the discussion a while back where not doing this actually led to existing bugs in the core framework.